README.md

```diff
@@ -1,45 +1,189 @@
 
-This will install all the jars built by the project.  Depend on the `sctp` module to use jitsi-sctp in your code.
+  * The `jnilib` maven module combines the `jniwrapper-java` and `jniwrapper-native` into a single `jar` which includes the Java API and the native JNI library that will be loaded at runtime.
+  When built with `mvn package -f pom.xml -Pbuild-usrsctp -Pbuild-jnisctp -pl org.jitsi:jnilib -am` the `jnilib` artifact only include native `jnisctp` library for current platform.
+  To have universal (**fat jar** or **cross-platform jar**) `jnilib` suitable to run on any supported platform it necessary to build and publish platform-specific `jniwrapper-native` artifacts for all supported platforms in advance and then build with `build-x-plat-jar` profile enabled via passing `-Pbuild-x-plat-jar` switch into Maven. For example, fat `jnilib` jar could be built with `mvn package -Pbuild-x-plat-jar -f pom.xml -pl org.jitsi:jnilib`, which will produce fat jar with example content:
+    ```
+    $ tree jnilib-1.0-SNAPSHOT --noreport
+    jnilib-1.0-SNAPSHOT
+    |-- META-INF
+    |   |-- MANIFEST.MF
+    |   `-- maven
+    |       `-- org.jitsi
+    |           |-- jnilib
+    |           |   `-- pom.xml
+    |           |-- jniwrapper-java
+    |           |   |-- pom.properties
+    |           |   `-- pom.xml
+    |           `-- jniwrapper-native
+    |               `-- pom.xml
+    |-- cz
+    |   `-- adamh
+    |       `-- utils
+    |           `-- NativeUtils.class
+    |-- lib
+    |   |-- osx-x86_64
+    |   |   `-- libjnisctp.jnilib
+    |   |-- linux-x86_64
+    |   |   `-- libjnisctp.so
+    |   |-- freebsd-x86_64
+    |   |   `-- libjnisctp.so
+    |   `-- windows-x86_64
+    |       |-- jnisctp.dll
+    |       `-- jnisctp.pdb
+    |-- native
+    |   `-- headers
+    |       `-- org_jitsi_modified_sctp4j_SctpJni.h
+    `-- org
+        `-- jitsi_modified
+            `-- sctp4j
+                |-- EightArgumentVoidFunc.class
+                |-- FourArgumentIntFunc.class
+                |-- IncomingSctpDataHandler.class
+                |-- OutgoingSctpDataHandler.class
+                `-- SctpJni.class
+    ```
+
+
+* The `sctp` module contains the Java library on top of the JNI code. 
+The jar built by this is what is intended to be used by other code.
+
+### Building the jar files
+* Clone the project and initialize [`usrsctp`](https://github.com/sctplab/usrsctp) git submodule.
+* Run `mvn package -f pom.xml` (and `mvn install -f pom.xml` to install locally). It assumes that platform specific artifacts like `usrsctp` and `jniwrapper-native` are already available in Maven. To compile and install all artifacts locally run `mvn install -Pbuild-usrsctp -Pbuild-jnisctp -f pom.xml`, this will build native and `Java` code for current platform and install resulting artifacts to local repository.
```

Comment:
I'm trying to follow these steps and I think we need some more detail here...I tried running `mvn package -f pom.xml` and got an error:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.2.0:single (make-assembly) on project jnilib: Failed to create assembly: Error adding file-set for 'org.jitsi:jniwrapper-native:jar:osx-x86_64:1.0-SNAPSHOT' to archive: Expected org.codehaus.plexus.components.io.resources.PlexusIoArchivedResourceCollection, got org.codehaus.plexus.components.io.resources.PlexusIoFileResourceCollection -> [Help 1]
```

Maybe it's because of the assumption you mention later?  I think we need to explain that assumption more, I don't follow it.

Reply:
@bbaldino I'll verify that latest version of PR is running on mac tomorrow. At some point I've checked compilation on all platforms, but then checked only on single platform. BTW, could you please try `mvn install -Pbuild-usrsctp -Pbuild-jnisctp -f pom.xml` instead of  just `mvn package -f pom.xml`?

Reply:
That did get further (I had to install cmake, we should put that on a list of pre-reqs, probably), but there were some test failures:

```
ERROR] Errors:
[ERROR]   Sctp4jTest.basicLoop:41 » UnsatisfiedLink org.jitsi_modified.sctp4j.SctpJni.us...
[ERROR]   Sctp4jTest.testInit:33 » UnsatisfiedLink org.jitsi_modified.sctp4j.SctpJni.usr...
```

Reply:
@bbaldino thanks for reporting! I'm able to reproduce this issue on mac as well. Seem I've broken native library loading for mac when added support for different linux architectures.

Reply:
@bbaldino fixed in latest commit. `mvn install -Pbuild-usrsctp -Pbuild-jnisctp -f pom.xml` succeed on my mac now. I'll fix readme tomorrow.

Reply:
Ok, we should do a multi-platform build github action probably for this

Reply:
@bbaldino I don't have experience with `GitHub Actions`, so it will probably take several days to implement build for `Linux`, `Windows`, `Mac` and `FreeBSD`.